### PR TITLE
Avoid dynamic mapping update in `test_types_with_storage_can_be_inserted_and_queried`

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

The `bit` type currently can't be created using dynamic mapping updates
Should fix flakyness.

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Review by matriv — APPROVED** [review]
Thank you!

**Comment by mfussenegger** [comment]
Still flaky and found what looks like another bug :/

**Comment by mfussenegger** [comment]
Had to change the typeDef logic a bit to handle objects within objects (It generated `x object as (x object (strict))` without the final child). And with https://github.com/crate/crate/pull/13212 merged this should now be good to go

File: server/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java [hunk 1]
```diff
@@ -1999,6 +2003,29 @@ public void test_bit_string_can_be_inserted_and_queried() throws Exception {
...
     }
 
+    /**
+     * Adds innerTypes to object type definitions
+     **/
+    private static DataType<?> extendedType(DataType<?> type, Object value) {
+        if (type.id() == ObjectType.ID) {
+            var entryIt = ((Map<?, ?>) value).entrySet().iterator();
+            Builder builder = ObjectType.builder();
+            while (entryIt.hasNext()) {
```
Comment (matriv) [inline server/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java:2013]: Out of curiosity: any reason to use iter instead of for loop?

**Review by matriv — APPROVED** [review]
LGTM, thx!

File: server/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java:2013]: It's a leftover from the first version which generated a string and used a .hasNext inside the loop to add a comma if it's not the last.

**Pull request merged**